tests/Behat/RdfEntityContext.php

```diff
@@ -0,0 +1,205 @@
+    if (empty($this->rdfEntities)) {
+      return;
+    }
+
+    foreach ($this->rdfEntities as $rdf) {
+      $rdf->delete();
+    }
+  }
+
+  /**
+   * Before scenario hook sets a default Provenance URI on the site.
+   *
+   * @param \Behat\Behat\Hook\Scope\BeforeScenarioScope $scope
+   *   The Hook scope.
+   *
+   * @BeforeScenario @rdf-provenance
+   */
+  public function setDefaultProvenance(BeforeScenarioScope $scope): void {
+    $this->configContext->setConfig('oe_content.settings', 'provenance_uri', 'http://example.com');
+  }
+
+  /**
+   * Configures the Provenance URI.
+   *
+   * @Given the site Provenance URI is set to :uri
+   */
+  public function theSiteProvenanceUriIsSetTo($uri): void {
+    $this->configContext->setConfig('oe_content.settings', 'provenance_uri', $uri);
+  }
+
+  /**
+   * User creation and login for creating RDF entities.
+   *
+   * A custom step that creates a user that has the permissions to create and
+   * view RDF entities of a give type.
+   *
+   * @Given I am logged in with a user that can create and view :bundle RDF entities
+   */
+  public function iAmLoggedInWithUserThatCanCreateAndViewRdfEntityTypes($bundle) {
+    /** @var \Drupal\rdf_entity\RdfEntityTypeInterface[] $types */
+    $types = \Drupal::entityTypeManager()->getStorage('rdf_type')->loadMultiple();
+    $permission_map = [];
+    foreach ($types as $id => $type) {
+      $permission_map[$type->label()] = "create $id rdf entity";
+    }
+
+    if (!isset($permission_map[$bundle])) {
+      throw new \InvalidArgumentException('The provided entity type is not correct.');
+    }
+
+    $permission = $permission_map[$bundle];
+    $permissions = ['view rdf entity', 'view rdf entity overview', $permission];
+    $role = $this->getDriver()->roleCreate($permissions);
+
+    // Create user.
+    $user = (object) [
+      'name' => $this->getRandom()->name(8),
+      'pass' => $this->getRandom()->name(16),
+      'role' => $role,
+    ];
+    $user->mail = "{$user->name}@example.com";
+    $this->userCreate($user);
+
+    // Assign the temporary role with given permissions.
+    $this->getDriver()->userAddRole($user, $role);
+    $this->roles[] = $role;
+
+    // Login.
+    $this->login($user);
+  }
+
+  /**
+   * Creates a test RDF entity.
+   *
+   * @Given I create an :bundle RDF entity with the name :name
+   */
+  public function iCreateTestRdfEntityWithTheName($bundle, $name): void {
+    $values = [
+      'bundle' => $bundle,
+      'label' => $name,
+    ];
+
+    $this->createRdfEntity($values);
+  }
+
+  /**
+   * Tests that the created RDF entity has the correct provenance URI.
+   *
+   * @Then the Provenance URI of the RDF entity with the name :name should be :provenance_uri
+   */
+  public function theProvenanceUriOfTheRdfEntityWithTheNameShouldBe($name, $provenance_uri): void {
+    $entities = \Drupal::entityTypeManager()->getStorage('rdf_entity')->loadByProperties(['label' => $name]);
```

Comment:
We do this multiple times. Let's create `protected function getRdfEntityByLabel($label)`.

Reply:
@brummbar I refactored this in the other ticket, will be done there.